File: guilib-api/src/main/java/net/developertobi/guilib/api/item/GuiItem.java
package net.developertobi.guilib.api.item;

import net.developertobi.guilib.api.gui.GuiController;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.inventory.ItemStack;

public interface GuiItem {

    ItemStack getItem();
    void setItem(ItemStack item);

    void runAction(GuiPos position, GuiItem guiItem, InventoryClickEvent event);

    void update(GuiController controller, Modification modification, Object... values);

    enum Modification {
        TYPE,
        DISPLAY_NAME,
        LORE,
        AMOUNT,
        INCREMENT,
        ENCHANTMENTS,
        GLOWING
    }
}

File: guilib-api/bin/main/net/developertobi/inventorylib/api/item/GuiItem.kt
package net.developertobi.inventorylib.api.item

import net.developertobi.inventorylib.api.inventory.GuiController
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.inventory.ItemStack

interface GuiItem {

    var item: ItemStack

    fun runAction(position: GuiPos, guiItem: GuiItem, event: InventoryClickEvent)

    fun update(controller: GuiController, modification: Modification, vararg values: Any)

    enum class Modification {
        TYPE,
        DISPLAY_NAME,
        LORE,
        AMOUNT,
        INCREMENT,
        ENCHANTMENTS,
        GLOWING;
    }
}

